react/features/breakout-rooms/functions.ts

```diff
@@ -30,37 +37,62 @@ export const getMainRoom = (stateful: IStateful) => {
     return _.find(rooms, room => Boolean(room.isMainRoom));
 };
 
+/**
+ * Returns the rooms info.
+ *
+ * @param {IStateful} stateful - The redux store, the redux.
+
+* @returns {IRoomsInfo} The rooms info.
+ */
 export const getRoomsInfo = (stateful: IStateful) => {
     const breakoutRooms = getBreakoutRooms(stateful);
-    const conference = getCurrentConference(stateful);
+    const conference: IJitsiConference = getCurrentConference(stateful);
 
     const initialRoomsInfo = {
         rooms: []
     };
 
     // only main roomn
     if (!breakoutRooms || Object.keys(breakoutRooms).length === 0) {
+        // filter out hidden participants
+        const conferenceParticipants = conference?.getParticipants()
+            .filter((participant: IJitsiParticipant) => !participant.isHidden());
+
+        const localParticipant = getLocalParticipant(stateful);
+        let localParticipantInfo;
+
+        if (localParticipant) {
+            localParticipantInfo = {
+                role: localParticipant.role,
+                displayName: localParticipant.name,
+                avatarUrl: localParticipant.loadableAvatarUrl,
+                id: localParticipant.id
+            };
+        }
+
         return {
             ...initialRoomsInfo,
             rooms: [ {
                 isMainRoom: true,
                 id: conference?.room?.roomjid,
                 jid: conference?.room?.myroomjid,
-                participants: conference?.participants && Object.keys(conference.participants).length
-                    ? Object.keys(conference.participants).map(participantId => {
-                        const participantItem = conference?.participants[participantId];
-                        const storeParticipant = getParticipantById(stateful, participantItem._id);
-
-                        return {
-                            jid: participantItem._jid,
-                            role: participantItem._role,
-                            displayName: participantItem._displayName,
-                            avatarUrl: storeParticipant?.loadableAvatarUrl,
-                            id: participantItem._id
-                        };
-                    }) : []
-            } ]
-        };
+                participants: conferenceParticipants?.length > 0
```

Comment:
Why do we get the participants from lib-jitsi-meet and not from redux in base/participants?

Reply:
If I remember correctly, I did it like this because we want `jid` as well in the results and from lib-jitsi-meet we have the method `isHidden()` which is used initially to filter out the hidden participants. The prop `jid` is not in the redux base/participants